### DISCUSSION TITLE
Unable to install and run Tor browser in my kali VM

### DISCUSSION THREAD

**Discussion by Alonewalker11** [utmapp/UTM#3167] [General]
### Description of the issue

I have a kali VM up and running inside UTM in my Mac M1.
I tried to install the Tor browser inside the VM but it just doesn't installs.
I went to the default Firefox browser > went to the official torprojects.org page >downloaded the linux version and saved the file > I extracted the file > then clicked on start-Tor-browser file...this should have started installation but install some errors occurred...I will attach some Screenshots of the same...

**Configuration**
* UTM Version: Version 2.2.4 (36)
* OS Version: Mac OS Big Sur 11.6 and kali-linux 5.10.0

<img width="1440" alt="Screenshot 2021-10-06 at 4 35 06 PM" src="https://user-images.githubusercontent.com/90711387/136259936-3604fc8d-2481-4328-8cbb-9c6a2a3d98cc.png">
<img width="1440" alt="Screenshot 2021-10-06 at 4 34 31 PM" src="https://user-images.githubusercontent.com/90711387/136259993-2e2beb9b-274a-4328-92c5-132f723a3364.png">

<img width="731" alt="Screenshot 2021-10-06 at 11 40 55 PM" src="https://user-images.githubusercontent.com/90711387/136260009-06e955ff-cf8e-4450-a014-d4ef143dd5d1.png">

**Comment by conath** [comment]
It seems like the Tor Browser is not yet available for ARM versions of linux. You are trying to run the x86 Tor Browser on an ARM Virtual Machine, resulting in the error. You'll need to wait for another Download to be made available on the Tor Project website, or install an x86_64 version of Kali in a new VM.

As this is not a technical problem with UTM, I am closing and moving this to Discussions.

**Reply by Alonewalker11** [reply]
thanks @conath .....but the x86_64 0ne is really slow...okay lets wait....I'm gonna put this issue in tor community...

**Reply by CologeroG** [reply]
Im currently going through the same issue with getting Tor on Kali UTM VM on my M1

**Reply by FrankVerse** [reply]
Hey I'm as well struggling on getting Tor up and running on my M1 , following up on the previous thread does anyone know if tor is now available for use in M1 UTM VM Kali Linux ?

**Comment by DanielSiebert-dev** [comment]
As per torrc the configuration is ok for the standard user :). I use the M2 with utm 4.5 and kali 2024.2. Maybe so was working on these headings (1-6). I will go a different way after 6:

1. Check Proxy Settings: 
Ensure that the proxy settings of Kali Linux are configured correctly to communicate with the UTM. This might involve configuring a specific proxy server and port to route outgoing network traffic through the UTM.

2. Firewall Settings: 
Review the firewall settings of the UTM to ensure that they allow outgoing Tor traffic. You may need to create specific rules to allow Tor traffic, especially if the UTM blocks outgoing traffic by default.

3. Port Forwarding: 
Ensure that the ports used by Tor (typically port 9050 for SOCKS5) are forwarded in the UTM to allow traffic from Kali Linux. This is important for Tor to establish connections to its network.

4. DNS Settings: 
Check the DNS settings of Kali Linux and ensure they are compatible with the DNS settings of the UTM. You may need to configure the DNS servers used by Tor to ensure that DNS queries are forwarded correctly.

5. Authentication and Encryption: 
Ensure that communication between Kali Linux and the UTM is properly authenticated and encrypted, especially if sensitive data is being transmitted over the network.

6. Network Topology: 
Consider the network topology of the UTM environment and ensure that Kali Linux is integrated into the network and can communicate with other resources, such as internal servers or services. So, for now it should be no more problem of UTM. I am trying to bring up a solution within kali.

Hope so finds a solution for that, soon.

I changed my settings in the utm to the following:
In the network menu of utm for mac my network mode is bridged with an interface of en0. the emulated network card is virtio-net-pci. i also use utm v 4.5 and isolate guest from host. The virtual machines can only communicate with the physical network and are fully isolated from each other. Therefore i only needed to check with kali itself and it worked.

I still recommend to use a proxy and vpn.
